pydocstringformatter/formatting/base.py

```diff
@@ -8,6 +9,8 @@ class Formatter:
     """Base class for docstring formatter."""
 
     optional = False
+    config: argparse.Namespace
+    """Namespace object set when set_config_namespace is called."""
```

Comment:
```suggestion
```
Maybe we could let child class create an attribute for the configuration of interest for them. Right now this configuration is a black box and you need to read a lot of code to see what's in it. If we have to create the attribute in the child class then you only have to read the child class.

Reply:
Hmm, I don't immediately see what the benefit would be here.

We need a function in which the `config` attribute of the `Run` class is made accessible to the formatters. We can then set `config` as attribute on the formatter or only set the necessary attributes. But both would still require you to understand what's in the `Run.config` attribute. I could see a `set_specific_config_attributes` method (although I wonder how beneficial that would be), but I don't immediately see how not setting `self.config` here makes it easier to interact with the `Namespace` object later on. At some point you will need to know what you can and cannot extract form it.

Reply:
The person that add the option has the entire context in mind when creating the option and setting it in their formatter here : https://github.com/DanielNoord/pydocstringformatter/pull/53#discussion_r807860798. By doing so they establish a clear list of relevant options for this formatter. The person reading the class to modify it later does not, but they can read ``set_config_namespace`` to get the full list of relevant option. The alternative for them is to have to read the full list of option in argparse and to ask oneself what option is relevant for the formatter. (Actually this is something I truly dislike about the way pylint handle option in checkers).

Reply:
Hm, the downside of this is that the value of the option becomes detached from the `Namespace`. If for whatever reason the `Namespace` object changes (for example if we want to support different `Namespaces` for different directories) then we would need to update all formatters again.
I see how it would provide some clarity to somebody looking at the class, but on the other hand just searching for all occurrences of `self.config` will show you all relevant options as well.

Reply:
> If for whatever reason the Namespace object changes (for example if we want to support different Namespaces for different directories) then we would need to update all formatters again.

True. The question then become : How often are we going to change the namespace object vs reading the formatter's code ?

> but on the other hand just searching for all occurrences of self.config will show you all relevant options as well.

Fair enough, it can become overwhelming for pylint because checkers are huge sometime 2000+ lines. Not a problem here I guess. I'll let you decide :)

Reply:
> > If for whatever reason the Namespace object changes (for example if we want to support different Namespaces for different directories) then we would need to update all formatters again.
> 
> True. The question then become : How often are we going to change the namespace object vs reading the formatter's code ?

Not sure, although I think accessing new options in a formatter shouldn't happen too often. Until now all formatters have been relatively concise.

> > but on the other hand just searching for all occurrences of self.config will show you all relevant options as well.
> 
> Fair enough, it can become overwhelming for pylint because checkers are huge sometime 2000+ lines. Not a problem here I guess. I'll let you decide :)

Agreed and it's a good point to think about during the migration to `argparse`. Although I don't have an easy solution right away.